Makefile

```diff
@@ -70,12 +70,13 @@ clean:
 .PHONY: clobber
 clobber: clean
 	rm -rf \
-	    nginx \
-		dd-opentracing-cpp/deps
+		nginx \
+		dd-opentracing-cpp/deps \
+		dd-opentracing-cpp/CMakeFiles
 
 .PHONY: build-in-docker
 build-in-docker: sources
-	bin/run_in_build_image.sh $(MAKE) BUILD_DIR=.docker-build build
+	bin/run_in_build_image.sh make BUILD_DIR=.docker-build build
```

Comment:
`$(MAKE)` is the GNU Make way of doing recursive make, per <https://www.gnu.org/software/make/manual/html_node/MAKE-Variable.html>.

You explained:

> passing $(MAKE) into run_in_build_image.sh ends up trying to run /Library/Developer/CommandLineTools/usr/bin/make in a Linux image, which fails

I wonder if the issue is the difference between GNU Make and non-GNU make. Maybe if instead of this change, we specified `gmake` everywhere we currently have `make`, it would work. What do you think?

Reply:
The two `make` binaries on my M1 MBP are both GNU Make:

```shell
$ /usr/bin/make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0

$ /Library/Developer/CommandLineTools/usr/bin/make -v
GNU Make 3.81
Copyright (C) 2006  Free Software Foundation, Inc.
This is free software; see the source for copying conditions.
There is NO warranty; not even for MERCHANTABILITY or FITNESS FOR A
PARTICULAR PURPOSE.

This program built for i386-apple-darwin11.3.0
```

Since the existing system already has GNU make, there's no gmake binary available on a default MacOS system or via homebrew.

Both `/usr/bin/make` and `/Library/Developer/CommandLineTools/usr/bin/make` expand `$(MAKE)` to `/Library/Developer/CommandLineTools/usr/bin/make`:

```shell
$ cat Makefile
default:
        echo $(MAKE)
$ /usr/bin/make
echo /Library/Developer/CommandLineTools/usr/bin/make
/Library/Developer/CommandLineTools/usr/bin/make
```

These two executables have different md5sums and inode numbers, so they're not the same file. The reason for the above behavior isn't clear to me, but I'm on a relatively clean MacOS install and this behavior is consistent across bash and zsh, so I believe this is the default behavior.

I'd also argue that what this particular script is doing isn't (or perhaps shouldn't be) truly recursive make, as it is executing `make` in what is essentially an entirely new userspace inside the build container.